### PR TITLE
refactor(module): use `extendBuild`

### LINKED ISSUE #5: Refactor to extendBuild
https://github.com/nuxt-community/separate-env-module/blob/master/lib/module.js#L37

### PR DESCRIPTION
**Pull request by ricardogobbosouza**
Resolves #5

### CODE REVIEW

File: test/module.test.js [hunk 1]
```diff
@@ -64,33 +64,6 @@ describe('module', () => {
     expect(html).toMatchSnapshot()
   })
 
-  test('honor custom build.extend function', async () => {
```
Comment (TheAlexLichter) [inline test/module.test.js]: I'd still keep that test

**Review by TheAlexLichter — CHANGES_REQUESTED** [review]

**Comment by ricardogobbosouza** [comment]
@manniL Actually I think we can not apply `extendBuild` here because the` build.extend` function is called before, ie its values are overwritten and in the case it should be the other way around

**Comment by TheAlexLichter** [comment]
@ricardogobbosouza Not sure why that should be a problem. Ideally, the `separate-env` module is called at the end so all env values that have been "created" before could be separated properly.

Or am I missing a point here?

**Comment by ricardogobbosouza** [comment]
@manniL If you look for the test `honor custom build.extend function`
https://github.com/nuxt-community/separate-env-module/blob/master/test/module.test.js#L85 should the `ONLY_SERVER` variable be` yup` or `nope`?

**Comment by TheAlexLichter** [comment]
@ricardogobbosouza `nope`

**Comment by ricardogobbosouza** [comment]
@manniL So we have a problem, why today `separate-env` is called first of all https://github.com/nuxt-community/separate-env-module/blob/master/lib/module.js#L36 with `extendBuild` he will be called in sequence

**Comment by ricardogobbosouza** [comment]
In my opinion we can not apply `extendBuild` here

**Comment by ricardogobbosouza** [comment]
@manniL In my opinion we can not apply `extendBuild` here

**Comment by ricardogobbosouza** [comment]
@manniL 

Today we can not define the `extendBuid` order
https://github.com/nuxt/nuxt.js/blob/dev/packages/core/src/module.js#L104

And in this case we need to do so:
```js
this.options.build.extend = chainFn(fn, this.options.build.extend)
```
The `separate-env` should be the first.

**Comment by TheAlexLichter** [comment]
Yeah, you are right :relaxed:

**Pull request closed without merging**